packages/data-stores/src/queries/use-launchpad.ts

```diff
@@ -69,14 +70,27 @@ export const fetchLaunchpad = (
 		  } as APIFetchOptions );
 };
 
+const mapTask = ( task: Task, index: number ) => {
+	task.order = index;
+	return task;
+};
+
 export const useLaunchpad = (
 	siteSlug: string | null,
 	checklist_slug?: string | 0 | null | undefined
 ) => {
 	const key = [ 'launchpad', siteSlug, checklist_slug ];
 	return useQuery( {
 		queryKey: key,
-		queryFn: () => fetchLaunchpad( siteSlug, checklist_slug ),
+		queryFn: () =>
+			fetchLaunchpad( siteSlug, checklist_slug ).then( ( data ) => {
+				const tasks = data.checklist || [];
+				const completedTasks = tasks.filter( ( task: Task ) => task.completed );
+				const incompleteTasks = tasks.filter( ( task: Task ) => ! task.completed );
+				data.checklist = [ ...completedTasks, ...incompleteTasks ].map( mapTask );
+
+				return data;
+			} ),
```

Comment:
I _really_ like the change to perform the sorting somewhere central like this, but I am concerned that not all uses of the launchpad APIs will want this specific sort. I think we should make the sorting controlled by a flag/option. If we can confirm that we perform the same sorting in the fullscreen launchpad and any other consumers of the API, then we can make the sorting the default, but I still think we should retain some flexibility in the hook.